assemble.py

```diff
@@ -100,24 +99,35 @@ def _resize_wic_file(self, increase_bytes: int, extra_space=0.2):
         subprocess.check_call(['parted', self._path, 'resizepart', str(self._last_part), '100%'])
 
 
-def copy_container_images_from_archive_to_wic(target: FactoryClient.Target, apps_archive_dir: str, app_preload_dir: str,
-                                              wic_image: str, token: str, apps_shortlist: list, progress: Progress):
-
+def copy_compose_apps_to_wic(target: FactoryClient.Target, fetch_dir: str, wic_image: str, token: str,
+                             apps_shortlist: list, progress: Progress):
     p = Progress(2, progress)
-    target_app_store = ArchiveTargetAppsStore(apps_archive_dir)
-    target.shortlist = apps_shortlist
-    if not target_app_store.exist(target):
-        logger.info('Container images have not been found, trying to obtain them...')
-        apps_fetcher = TargetAppsFetcher(token, app_preload_dir)
-        apps_fetcher.fetch_target_apps(target, apps_shortlist)
-        apps_fetcher.fetch_apps_images()
-        target_app_store.store(target, apps_fetcher.target_dir(target.name))
+    apps_fetcher = TargetAppsFetcher(token, fetch_dir)
```

Comment:
two new calls to `p.tick()` have been added here. I think line 104 needs to be changed to `p = Progress(4, progress)`